packages/core-container/lib/registrars/plugin.js

```diff
@@ -57,10 +57,12 @@ module.exports = class PluginRegistrars {
    * @return {void}
    */
   async tearDown () {
-    const plugins = this.deregister.reverse()
+    const modules = this.deregister.reverse()
 
-    for (let i = 0; i < plugins.length; i++) {
-      await plugins[i].plugin.deregister(this.container, plugins[i].options)
+    for (let i = 0; i < modules.length; i++) {
+      if (modules[i].plugin.deregister) {
```

Comment:
This 'if' block here is un-necessary? Modules only end up in the 'this.deregister' array if they have a 'deregister' property set

Reply:
Good catch!